scene/3d/physics_body.cpp

```diff
@@ -1018,8 +1018,12 @@ Vector3 KinematicBody::move_and_slide(const Vector3 &p_linear_velocity, const Ve
 		}
 	}
 
-	// Hack in order to work with calling from _process as well as from _physics_process; calling from thread is risky
-	Vector3 motion = (floor_velocity + body_velocity) * (Engine::get_singleton()->is_in_physics_frame() ? get_physics_process_delta_time() : get_process_delta_time());
+	if (delta < 0.0) {
```

Comment:
Can you please use `<=` we have to avoid the possibility to send 0 as delta to avoid undefined behaviour.

Reply:
I originally had that, but was asked to change it by @bojidar-bg in a previous comment. I agree with you though, it should probably protect against 0. I will change it back.

Reply:
@AndreaCatania I've made the change you have requested. Let me know if there is anything else. :)

Reply:
Perfect, thanks! I've approved it.